bin/systemml-spark-submit.py

```diff
@@ -21,167 +21,120 @@
-             '--num-executors', args.num_executors, '--executor-memory', args.executor_memory,
-             '--executor-cores', args.executor_cores, '--conf', conf, '--jars', target_jars,
-             systemml_jar]
-
-cmd_system_ml = ['-config', systemml_config_path_arg,
-                 '-exec', vars(args)['exec'], '-f', script_file, ' '.join(ml_options)]
+    spark_home, systemml_home = get_env()
+    spark_path = join(spark_home, 'bin', 'spark-submit')
+    script_file = find_script_file(systemml_home, f)
 
-cmd = cmd_spark + cmd_system_ml
+    # Jars
+    cuda_jars, systemml_jars = default_jars(systemml_home)
 
-return_code = os.system(' '.join(cmd))
-# For debugging
-# print(' '.join(cmd))
+    # Log4j
+    log4j = log4j_path(systemml_home)
+    log4j_properties_path = 'spark.driver.extraJavaOptions=-Dlog4j.configuration=file:{}'.format(log4j)
+    if conf is None:
+        default_conf = log4j_properties_path
+    else:
+        default_conf = ' --conf '.join(conf + [default_conf])
 
-if return_code != 0:
-    print('Failed to run SystemML. Exit code :' + str(return_code))
-    print(' '.join(cmd))
+    # Config
+    if config is None:
+        default_config = config_path(systemml_home)
+    else:
+        default_config = ' --conf '.join(conf + [default_conf])
+
+    #  optional arguments
+    ml_options = []
+    if nvargs is not None:
+        ml_options.append('-nvargs')
+        ml_options.append(' '.join(nvargs))
+    if args is not None:
+        ml_options.append('-args')
+        ml_options.append(' '.join(args))
+    if explain is not None:
+        ml_options.append('-explain')
+        ml_options.append(explain)
+    if debug is not False:
+        ml_options.append('-debug')
+    if stats is not None:
+        ml_options.append('-stats')
+        ml_options.append(stats)
+    if gpu is not None:
+        ml_options.append('-gpu')
+        ml_options.append(gpu)
+
+    if len(ml_options) < 1:
+        ml_options = ''
+
+    # stats, explain, target_jars
+    cmd_spark = [spark_path, '--class', 'org.apache.sysml.api.DMLScript',
+                 '--master', master, '--driver-memory', driver_memory,
+                 '--num-executors', num_executors, '--executor-memory', executor_memory,
+                 '--executor-cores', executor_cores, '--conf', default_conf,
+                 '--jars', cuda_jars, systemml_jars]
+
+    cmd_system_ml = ['-config', default_config,
+                     '-exec', 'hybrid_spark', '-f', script_file, ' '.join(ml_options)]
+
+    cmd = cmd_spark + cmd_system_ml
+    return_code = os.system(' '.join(cmd))
+    return return_code
+
+
+if __name__ == '__main__':
+    spark_home, systemml_home = get_env()
+
+    cparser = argparse.ArgumentParser(description='System-ML Spark Submit Script')
+    # SPARK-SUBMIT Options
+    cparser.add_argument('--master', default='local[*]', help='local, yarn-client, yarn-cluster', metavar='')
```

Comment:
Can you please also print out the defaults for each of the options in the help message?